lib/ddtrace/vendor/redis/connection_specification.rb

```diff
@@ -0,0 +1,163 @@
+require 'uri'
+
+# NOTE: This code is copied directly from Redis.
+#       Its purpose is to resolve connection information.
+#       It exists here only because it doesn't exist in the redis
+#       library as a separated module and it allows to avoid
+#       instantiating a new Redis::Client for resolving the connection
+module Datadog
+  module Vendor
+    module Redis
+      # The following module mirrors what has been done for ActiveRecord
+      # connection specification
+      module ConnectionSpecification
+        ##
+        # Builds a ConnectionSpecification from user input.
+        class Resolver # :nodoc:
```

Comment:
Although copying code from other gems is usually not a good idea, I think this makes practical sense for the sake of stability and performance. These options are unlikely to change anytime soon, and we should be able to reap the benefits for the foreseeable future.

My only suggestion here might be to move the namespaces a bit, in an effort to compartmentalize the integrations. 

For Rails, we had it in `ddtrace/vendor` because it was a direct copy paste, it is deprecated and intended to be removed when we drop support for older Rails, and because our gem structure at the time was more tightly coupled integrations with the core.

However, I think for Redis here, it might be better to put this in `Datadog::Contrib::Redis::Vendor::Resolver` or at the minimum ensure any copied code lives in `ddtrace/contrib/redis/vendor`, such that `ddtrace/contrib/redis` becomes a totally complete package of all the Redis integration code. Frankly, Rails should also be updated to such a standard as well (move the code to `ddtrace/contrib/active_record/vendor`), but that's not required of this PR.

Reply:
I moved the vendor resolver into `lib/ddtrace/contrib/redis/vendor/resolver.rb`